### DISCUSSION TITLE
Voice instructions language

### DISCUSSION THREAD

**Discussion by mosherif21** [eopeter/flutter_mapbox_navigation#275] [Q&A]
First of all i want to thank you for the amazing package that saved me a lot of time, i have a little problem when changing the language of the regular navigation the problem is when i change the language to Arabic the streets text, voice instructions text and turns text change to Arabic but the voice instruction audio that's played language doesn't change so it tries to convert the Arabic text of the instructions to speech but in English so the words come out wrong and not recognizable. and is there a way to change the measuring units from miles and ft to Kilometre

**Comment by AndrewPearce** [comment]
Hello Mosherif21.
From the docs the way to change the measuring units from miles to kilometres is to use the units parameter ie
units: VoiceUnits.imperial,
or 
untis: VoiceUnits.metric,
But changing this option doesn't make any difference for me, the navigation widget still shows distances in miles and the voice uses miles as well. Anyone else have this problem ? Anyone found a solution ?

**Comment by franciscosouzti** [comment]
Mapbox has a setting to set the language, but when trying to put Portuguese (Brazil) it doesn't work

**Comment by eopeter** [comment]
> First of all i want to thank you for the amazing package that saved me a lot of time, i have a little problem when changing the language of the regular navigation the problem is when i change the language to Arabic the streets text, voice instructions text and turns text change to Arabic but the voice instruction audio that's played language doesn't change so it tries to convert the Arabic text of the instructions to speech but in English so the words come out wrong and not recognizable. and is there a way to change the measuring units from miles and ft to Kilometre
> 
> 

Yes, use the VoiceUnits with value metric
